.github/workflows/spanner-pr.yml

```diff
@@ -0,0 +1,201 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Checks that are intended to run on PRs containing Java code.
+
+name: Spanner PR
+
+on:
+  pull_request:
+    branches:
+    - 'main'
+    paths:
+      # Include spanner paths only
+      - '.github/workflows/spanner-pr.yml'
+      - 'v2/datastream-to-spanner/**'
+      - 'v2/spanner-common/**'
+      - 'v2/spanner-change-streams-to-sharded-file-sink/**'
+      - 'v2/gcs-to-sourcedb/**'
+      - 'v2/spanner-migrations-sdk/**'
+      - 'v2/spanner-custom-shard/**'
+      - 'v2/sourcedb-to-spanner/**'
```

Comment:
I would recommend including `v2/common/**` here as well since changes there could break spanner

Reply:
The default path builds all the templates so this would be caught irrespective of who (Spanner team or someone else) did the change. Do you think we still need to add it?

Reply:
That's a good point, I think you're fine without it. I also added this comment before I saw the `-am` change (which reduces the dependency here).